internal/audit/hub/hub.go

```diff
@@ -268,16 +271,24 @@ func (l *Log) streamPrefix(ctx context.Context, kind logsv1.IngestBatch_EntryKin
 		kv := kvPool.Get().(*bpb.KV) //nolint:forcetypeassert
 		defer kvPool.Put(kv)
 
+		var keys [][]byte
+		if keysIface := keysPool.Get(); keysIface == nil {
+			keys = make([][]byte, l.maxBatchSize)
+		} else {
+			keys = *(keysIface.(*[][]byte))
+		}
+		defer keysPool.Put(&keys)
+
 		var i int
-		keys := make([][]byte, l.maxBatchSize)
 		if err := buf.SliceIterate(func(s []byte) error {
 			kv.Reset()
 
 			if err := kv.Unmarshal(s); err != nil {
 				return err
 			}
 
-			keys[i] = append([]byte(nil), kv.Key...)
+			clear(keys[i])
```

Comment:
If this is a newly created object from the pool `keys[i]` would be nil and would require instantiation first.

```suggestion
if keys[i] == nil {
  keys[i] = make([]byte, len(kv.Key))
} else {
  clear(keys[i])
}
```